src/forms.js

```diff
@@ -32,11 +32,10 @@ fn.extend({
   },
 
   val(value) {
-    if (value === undefined) {
-      return this[0].value;
-    } else {
-      return this.each(v => v.value = value);
-    }
+    return ( value === undefined ?
+      this[0].value :
+      this.each(v => v.value = value)
+    );
   }
```

Comment:
That's a big driver behind wanting to document the code. :-)

Since one of cash's selling points is the small size, there are instances
where we have to get creative to save bytes. Individual bytes don't matter
much, but those design decisions can mean the difference between 12kb and
9kb in the final file.

As you mention, I'm not sure if this is one of those cases, since a good
minifier should handle that automatically. My change there was for
consistency, since most of the similar methods are in that format for
ternary returns.

Overall, I try never to go more than one ternary deep `( condition ?
condition2 ? 'Yes' : 'No' : 'Maybe' )` since that is absolutely hard to
parse and I break it out on to multiple lines to help with readability.

We definitely should take a pass over everything to update for consistency
and establish a few standards.
On Sat, May 7, 2016 at 4:05 AM Simon Goellner <notifications@github.com>
wrote:

> In src/forms.js
> <https://github.com/kenwheeler/cash/pull/116#discussion_r62414198>:
>
> > @@ -32,11 +32,10 @@ fn.extend({
> >    },
> >
> >    val(value) {
> > -    if (value === undefined) {
> > -      return this[0].value;
> > -    } else {
> > -      return this.each(v => v.value = value);
> > -    }
> > +    return ( value === undefined ?
> > +      this[0].value :
> > +      this.each(v => v.value = value)
> > +    );
> >    }
>
> to build on this... I know You, Ken and such are pretty hardcore JS
> developers and can probably read this kind of code like it's a children's
> book... but personally storing ternary structures in my brain is
> difficult...
>
> It also worries me that any regular joes like myself who use Cash but find
> bugs, or difficulties and want to contribute will be pretty put-off if htey
> can't debug the un-minified code themselves.
>
> Perhaps if shaving bytes is a particularly large goal, we can investigate
> ways to compress the code even better with tools, rather than replacing all
> our expressions with ternaries ? :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kenwheeler/cash/pull/116/files/3daffdf54d6f1474306f6d055e999651c981d094..d7db714e55e2b9f50ac733ecafc5011e0d97fc1c#r62414198>
>